compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2532,7 +2540,7 @@ object Parsers {
               if (leading == LBRACE || in.token == CASE)
                 enumerators()
               else {
-                val pats = patternsOpt()
+                val pats = patternsOpt(EMPTY)
```

Comment:
Why `EMPTY` in this case? Shouldn't it be something like OUTDENT ?

Reply:
I thought this was to handle `for x, y in (1,2) do` or `for x, y in (1,2) yield`, so I think the expected end would have to be `do` or `yield`? Passing `EMPTY` preserves the old behavior, so I figured it was safe. I could pass predicate down instead I suppose. What do you think?

Reply:
I guess this if fine, the other possibility would be an `Option` but I don't see it used anywhere in the class, so probably there is a reason for it.

Reply:
I was maybe being overly cautious but I didn't want to have a perf hit from creating a lot of unnecessary `Some` wrappers.